src/ContractInteractor.ts

```diff
@@ -919,20 +945,25 @@ export default class ContractInteractor {
 
     async getERC20Token(
         address: string,
-        options?: TokenOptions
-    ): Promise<Token> {
-        const token = await this._createERC20(address);
+        options?: Partial<ERC20Options>
+    ): Promise<ERC20Token> {
+        const instance = await this._createERC20(address);
+        if (!options || Object.values(options).every((x) => x === false)) {
+            return { instance };
+        }
         const [name, decimals, symbol] = await Promise.all([
-            await token.name(),
-            options?.decimals ? (await token.decimals()).toNumber() : undefined,
-            options?.symbol ? await token.symbol() : undefined
+            options.name ? await instance.name() : undefined,
+            options.decimals
+                ? (await instance.decimals()).toNumber()
+                : undefined,
+            options.symbol ? await instance.symbol() : undefined
         ]);
```

Comment:
why using promise.all?

```suggestion
const name = options.name ? await instance.name() : undefined;
const name = options. decimals ? await instance. decimals() : undefined;
const name = options. symbol ? await instance. symbol() : undefined;
```

tbh though, this system seems more complicated than it needs to be. perhaps, if flags were typed as:
```
type OptionsFlags<Type> = {
    [Property in keyof Type]: true | undefined;
};
```
then you could just do:
```suggestion
const name = options.name && await instance.name();
const name = options. decimals && await instance. decimals();
const name = options. symbol && await instance. symbol();
```
to make it more readable.

Reply:
The promise.all was used to query multiple instances at the same time and not one by one.

Reply:
Indeed, I'd prefer to use `promise.all`. I don't see why we should retrieve the properties sequentially while we could retrieve them in parallel.